Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM --platform=$BUILDPLATFORM golang:1.24-alpine AS nobled-builder
+
+RUN apk add --no-cache \
+    build-base \
+    git \
+    linux-headers
+
+WORKDIR /src
+
+COPY . .
```

Comment:
To cache go deps we should be able to split this up into two steps. I remember doing this in Heighliner.  (https://github.com/strangelove-ventures/heighliner/blob/b73c4a259a59640c3fa4aa679ab0dd0b74d8c6e6/dockerfile/cosmos/local.Dockerfile#L27)

```Dockerfile
COPY go.mod go.sum ./

RUN go mod download
```

Then, since all the deps are cached, we can now copy the rest:

```Dockerfile
COPY . .
```